app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt

```diff
@@ -475,12 +481,39 @@ private fun MessageContent(
                     messageBody = messageContent.messageBody,
                     isAvailable = !message.isPending && message.isAvailable,
                     onLongClick = onLongClick,
-                    onOpenProfile = onOpenProfile
+                    onOpenProfile = onOpenProfile,
+                    buttonList = null,
+                    onButtonClick = null,
+                    pendingButton = null,
+                    messageId = message.header.messageId
                 )
                 PartialDeliveryInformation(messageContent.deliveryStatus)
             }
         }
 
+        is UIMessageContent.Composite -> {
+            Column {
+                messageContent.messageBody?.quotedMessage?.let {
+                    VerticalSpace.x4()
+                    when (it) {
+                        is UIQuotedMessage.UIQuotedData -> QuotedMessage(it)
+                        UIQuotedMessage.UnavailableData -> QuotedUnavailable(QuotedMessageStyle.COMPLETE)
+                    }
+                    VerticalSpace.x4()
+                }
+                MessageBody(
+                    messageBody = messageContent.messageBody,
+                    isAvailable = !message.isPending && message.isAvailable,
+                    onLongClick = onLongClick,
+                    onOpenProfile = onOpenProfile,
+                    buttonList = messageContent.buttonList,
+                    onButtonClick = onMessageButtonClicked,
+                    messageId = message.header.messageId,
+                    pendingButton = pendingButtonsMap[message.header.messageId]
+                )
+            }
+        }
```

Comment:
This whole thing could be a Composable on its own, that would containt this Column plus this 
```

 buttonList?.also {
        MessageButtonsContent(
            messageId = messageId,
            buttonList = it,
            onClick = onButtonClick,
            pendingButton = pendingButton
        )
    }
}
```